examples/InstrumentingLibraries/README.md

```diff
@@ -0,0 +1,26 @@
+# Instrumenting libraries with OpenTelemetry
+
+Included here are two examples of instrumenting libraries with OpenTelemetry.

```

Comment:
Do we want to document how to instrument a library with DiagnosticSource? its not the current recommendation, and its already documented in .NET Repo.

Also the ActivitySource document should also live in .NET Repo - but good to host it here for now, and we can submit a PR to .NET repo to make it part of their document..

(I am bit doubtful about the SupressInstrumentation part.. It could be documented separately, so that the library dont have to take any dependency to anything but DiagnosticSource.)

Reply:
Agreed we likely do not want either of these long term - especially the legacy example. This is partly the reason I opened this as a draft. Also, agreed that suppress instrumentation can and should be independently documented.

These are my motivations for this PR

1. We have community members either trying to use `ActivitySource` or are not yet aware of it. Here's a [gitter conversation](https://gitter.im/open-telemetry/opentelemetry-dotnet?at=5f2e976f51bb7d3380da447d) with a user trying to instrument a library with DiagnosticSource - I suggested `ActivitySource`. I agree that when there is good official documentation, there is no reason for an example to be maintained in this repository.
2. Regarding the suppress instrumentation functionality. Here's an [example](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1202#issuecomment-683477660) where someone is trying to use it, but running into issues. I'm interested in how my trivial example here differs from what he is experiencing.
3. Regarding the DiagnosticSource example, agreed we don't really need this at all, as there are plenty of examples in the repository. My motivation here is that I think it provides a simpler example to vet out the suppress instrumentation functionality with a DiagnosticSource instrumented library. My gut says there are going to be some tricky things to make this work and at least for me, this simple example makes it easier to vet things out.